experimental/cpu_ukernel/plugin.c

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/builtins/ukernel/api.h"
+#include "iree/hal/local/executable_plugin.h"
+
+// Implementation of iree_uk_assert_fail failure is deferred to users code, i.e.
+// to us here, as core ukernel/ code can't use the standard library.
+#if defined(IREE_UK_STANDALONE)  // Building a standalone plugin.
+void iree_uk_assert_fail(const char* file, int line, const char* function,
+                         const char* condition) {
+  // Doing nothing at the moment.
+}
+#else  // Building a system plugin.
+#include <stdio.h>
+#include <stdlib.h>
+void iree_uk_assert_fail(const char* file, int line, const char* function,
+                         const char* condition) {
+  fflush(stdout);
```

Comment:
no need to touch stdout here
```suggestion
```

Reply:
It's fairly standard practice and I thought I understood a simple rationale for it: if anything earlier logged to stdout, it would be best to see that before the assertion, no?

I was just copying the `iree_uk_assert_fail` code we have in 2 other places. If this should be changed, we can try to do that consistently.